--- a/Makerules
+++ b/Makerules
@@ -122,7 +122,7 @@
   endif
   LIB_CFLAGS = -fPIC
   ifeq ($(OS),Darwin)
-    LIB_LDFLAGS = -dynamiclib
+    LIB_LDFLAGS = -dynamiclib -Wl,-install_name,@rpath/$(notdir $@)
   else ifeq ($(OS),wasm)
     LIB_LDFLAGS = -shared -sSIDE_MODULE
     EXE_LDFLAGS = -sMAIN_MODULE
@@ -193,7 +193,7 @@
 endif
 
 # Default system libraries
-SYS_FREETYPE_LIBS := -lfreetype2
+SYS_FREETYPE_LIBS := -lfreetype
 SYS_HARFBUZZ_LIBS := -lharfbuzz
 SYS_JBIG2DEC_LIBS := -ljbig2dec
 SYS_JPEGXR_LIBS := -ljpegxr
@@ -216,10 +216,6 @@
   HAVE_GLUT := yes
   SYS_GLUT_CFLAGS := -Wno-deprecated-declarations
   SYS_GLUT_LIBS := -framework GLUT -framework OpenGL
-  CC = xcrun cc
-  AR = xcrun ar
-  LD = xcrun ld
-  RANLIB = xcrun ranlib
 
   ifneq ($(ARCHFLAGS),)
     $(warning "MacOS with ARCHFLAGS set. Assuming we are building for arm64, and setting HAVE_LIBCRYPTO to no.")
