From 8593a59d32a241c907f8ff623a8ba8960e00707e Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 9 Apr 2024 16:30:40 -0500
Subject: [PATCH] Update __main__.py

---
 scripts/wrap/__main__.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scripts/wrap/__main__.py b/scripts/wrap/__main__.py
index ddd23a2815..3c689bee91 100644
--- a/scripts/wrap/__main__.py
+++ b/scripts/wrap/__main__.py
@@ -1649,8 +1649,8 @@ def fn(name):
                     cpp_files_text = ''
                     for i in cpp_files:
                         cpp_files_text += ' ' + os.path.relpath(i)
-                    libmupdfcpp = f'{build_dirs.dir_so}/libmupdfcpp.so{so_version}'
-                    libmupdf = f'{build_dirs.dir_so}/libmupdf.so{so_version}'
+                    libmupdfcpp = f'{build_dirs.dir_so}/libmupdfcpp.dylib'
+                    libmupdf = f'{build_dirs.dir_so}/libmupdf.dylib'
                     if pyodide:
                         # Compile/link separately. Otherwise
                         # emsdk/upstream/bin/llvm-nm: error: a.out: No such
@@ -2226,8 +2226,8 @@ def fn(name):
                                 )
                         if command_was_run:
                             macos_patch( out_so,
-                                    f'{build_dirs.dir_so}/libmupdf.dylib{so_version}',
-                                    f'{build_dirs.dir_so}/libmupdfcpp.so{so_version}',
+                                    f'{build_dirs.dir_so}/libmupdf.dylib',
+                                    f'{build_dirs.dir_so}/libmupdfcpp.dylib',
                                     )
             else:
                 raise Exception( 'unrecognised --build action %r' % action)
